src/test/java/org/openmrs/contrib/qaframework/test/AddDiagnosisToVisitNoteTest.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.test;
+
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.contrib.qaframework.helper.BuildTests;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.ActiveVisitsPage;
+import org.openmrs.contrib.qaframework.page.ClinicianFacingPatientDashboardPage;
+import org.openmrs.contrib.qaframework.page.VisitNotePage;
+
+public class AddDiagnosisToVisitNoteTest extends LocationSensitiveApplicationTestBase {
+	
+	private TestData.PatientInfo patient;
+	
+	@Before
+	public void setup() {
+		patient = createTestPatient();
+		createTestVisit();
+	}
+	
+	@Test
+	@Category(BuildTests.class)
+	public void addDiagnosisToVisitNoteTest() {
+		ActiveVisitsPage activeVisitsPage = homePage.goToActiveVisitsSearch();
+		activeVisitsPage.search(patient.identifier);
+		
+		ClinicianFacingPatientDashboardPage patientDashboardPage = activeVisitsPage.goToPatientDashboardOfLastActiveVisit();
+		VisitNotePage visitNotePage = patientDashboardPage.goToVisitNote();
+		visitNotePage.enterDiagnosis("Pne");
+		visitNotePage.enterSecondaryDiagnosis("Bleed");
+		assertEquals("Pneumonia", visitNotePage.primaryDiagnosis());
+		assertEquals("Bleeding", visitNotePage.secondaryDiagnosis());
+		
+		patientDashboardPage = visitNotePage.save();
+		
+		List<String> diagnoses = patientDashboardPage.getDiagnoses();
+		assertThat(diagnoses, hasItems("Pneumonia", "Bleeding"));
+	}
+	
+	@After
+	public void tearDown() throws Exception {
+		deletePatient(patient);
+	}
+	
+	private void createTestVisit() {
+		new TestData.TestVisit(patient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();
+	}
+}
```

Comment:
instead of invoking `createTestVisit()` within the setUp() method, we can rather write its implementation within the @Before method and this will improve code quality. This applies to all other test classes with similar implementation!

Reply:
Let us keep things the way they are and we will refactor everything in a new pull request, Am requesting

Reply:
any reason for not doing everything at once and opting to create tickets to fix this?

Reply:
Lets keep things the way they were created from distro, then we will later refactor these selenium tests.

Reply:
does it cost any thing if the tests are refactored before the commit is merged?

Reply:
Am actually not getting what you are trying to suggest here , do you mind adding more pointers, to my understanding I see no difference.

Reply:
```js
@Before
    public void setup() throws Exception {
        patient = createTestPatient();
        new TestData.TestVisit(patient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();
    }

    @Test
    @Category(BuildTests.class)
    public void addDiagnosisToVisitNoteTest() {
    
    }

    @After
    public void tearDown() throws Exception {
        deletePatient(patient);
    }
}
``

Reply:
@kdaud  Thanks for suggestion however i do believe this does the same thing. And again for simplicity and understanding of code workflow. This very method will do us good with clear java approach.

Reply:
having everything defined within setUp() method would be the better approach and this should apply in all other test classes!

Reply:
Thanks, i think you are simply complicating things here. We have used this such kind of approach for almost the entire project.Let me be little detailed. 

Defining a method and simply calling it somewhere makes the code more clean and readable probably understanding.